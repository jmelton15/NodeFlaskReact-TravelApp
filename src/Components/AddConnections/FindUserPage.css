.FindUserPage-MainContainer {
    margin-top: 30px;
    /* background-color: #dab681; */
    justify-content: center;
    width: 80% !important;
    border-radius: 25px;
    align-items: center;
}
