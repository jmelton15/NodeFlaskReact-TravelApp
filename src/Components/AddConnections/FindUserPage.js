import "./FindUserPage.css";
import FindUserForm from "./FindUserForm";
import { NodeApi } from "../../APIRequests/nodeApi";
import { useState } from "react";
import ConnectionCard from "./ConnectionCard";
import { Redirect } from "react-router-dom";
import GetScreenWidth from '../../helpers/GetScreenWidth';


const FindUserPage = ({user,token}) => {
    const [screenWidth] = GetScreenWidth();
    const [foundUsers,setFoundUsers] = useState();

    if(!token || !user) {
        return <Redirect to="/"></Redirect>
    }
    
    const findUser = async (formData) => {
        let foundUsersData = await NodeApi.findUser(formData.username,token);
        console.log(foundUsersData)
        if(foundUsersData) setFoundUsers(foundUsersData);
    }
    if(screenWidth <= 600) {
        return (
            <div className="FindUserPage-MainContainer">
            <div>
                <FindUserForm findUser={findUser} user={user}/>
            </div>
            <div className="d-flex justify-content-center mx-3 mt-3">
            {foundUsers && foundUsers.map((connection) => {
                return <ConnectionCard connection={connection} user={user} token={token}/>
            })}
            </div>
        </div>
        )
    }
    else {
        return (
            <div className="container FindUserPage-MainContainer d-flex flex-column">
                <div>
                    <FindUserForm findUser={findUser} user={user}/>
                </div>
                <div className="container-sm w-50">
                {foundUsers && foundUsers.map((connection) => {
                    return <ConnectionCard connection={connection} user={user} token={token}/>
                })}
                </div>
            </div>
        )
    }
}

export default FindUserPage;