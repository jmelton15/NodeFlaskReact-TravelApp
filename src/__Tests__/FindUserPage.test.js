import { render } from '@testing-library/react';
import {user} from "./TestHelperObjects";
import FindUserPage from "../Components/AddConnections/FindUserPage";

// simple smoke test for FindUserPage component
it("renders a FindUserPage component without crashing", function() {
    render(<FindUserPage user={user} token={true} />);
});

// simple snapshot test for FindUserPage Component
it("should match snapshot for FindUserPage", function() {
    const {asFragment} = render (<FindUserPage user={user} token={true} />);
    expect(asFragment()).toMatchSnapshot();
});