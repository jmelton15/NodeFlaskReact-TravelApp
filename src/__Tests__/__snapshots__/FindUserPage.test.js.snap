// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for FindUserPage 1`] = `
<DocumentFragment>
  <div
    class="container FindUserPage-MainContainer d-flex flex-column"
  >
    <div>
      <form
        class="FindUserForm"
      >
        <div
          class="FindUserForm-FindUserFormGroup form-group"
        >
          <input
            class="form-control"
            id="username-searchbox"
            name="username"
            placeholder="Type Username To Find User..."
            type="text"
            value=""
          />
          <button
            class="mt-2 btn-lg btn btn-secondary"
            id="findUserBtn"
          >
            Find
          </button>
        </div>
      </form>
    </div>
    <div
      class="container-sm w-50"
    />
  </div>
</DocumentFragment>
`;
